daemon/cmd/daemon_main.go

```diff
@@ -869,6 +869,10 @@ func initializeFlags() {
 	flags.Duration(option.DNSProxyConcurrencyProcessingGracePeriod, 0, "Grace time to wait when DNS proxy concurrent limit has been reached during DNS message processing")
 	option.BindEnv(Vp, option.DNSProxyConcurrencyProcessingGracePeriod)
 
+	flags.Int(option.DNSProxyLockSize, 128, "Array size containing mutexes which protect against parallel handling of DNS response IPs")
```

Comment:
nit: It's more like a lock count rather than a size, but I see you're framing it in terms of an array size. Either way it doesn't matter so much since it's marked hidden and is a super-user tweaking option.

Reply:
I like lock count as a name, so will use that.